test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -27,7 +27,8 @@ import zio.{ Cause, Exit, ZIO }
  */
 final class Assertion[-A] private (
   val render: Assertion.Render,
-  val run: (=> A) => AssertResult
+  val run: (=> A) => AssertResult,
+  val expected: Option[Any] = None
```

Comment:
Really I think we want to say that the expected value has to be an `A` but we can't easily do that here because `A` is contravariant. What if we had `Assertion` take a `renderDiff: Option[A => String]` parameter (maybe returning a more structured data type than a string). Then `A` would be in contravariant position.

Reply:
I guess `renderDiff` could return `Vector[DiffElement]`, or some case class wrapping it.
In this case `Assertion.equalTo`  (or `assertEqualString`) will be the one deciding on diff implementation, not `DefaultTestReporter`

Reply:
We could do `diffOption: Option(A => Differ => String)` to defer the implementation of the differ to the test reporter.